apps/chat/src/components/Common/ApplicationDialog.tsx

```diff
@@ -647,35 +544,55 @@ export const ApplicationDialog = ({
                 <span className="ml-1 inline text-accent-primary">*</span>
               </label>
               <input
-                name="completionUrl"
+                {...register('completionUrl', {
+                  required: 'Completion URL is required.',
+                  validate: (value) => {
+                    try {
+                      new URL(value);
```

Comment:
I don'r really get why do we need `new URL(value);` along with `try-catch`

Reply:
new URL(value) is used for validating if a given value is a valid URL. It throws an exception if the value is not a valid URL which is then caught and handled in the catch block. Thus, it's a quick way to check a value for URL validity and handle any invalid URL entries. Regular expressions may not always exactly match all valid URLs, I added an extra check to be safe.